haystack/nodes/prompt/prompt_node.py

```diff
@@ -708,56 +649,110 @@ def set_default_prompt_template(self, prompt_template: Union[str, PromptTemplate
+    def get_prompt_templates(self) -> List[PromptTemplate]:
         """
         Returns the list of supported prompt templates.
         :return: List of supported prompt templates.
         """
-        return list(cls.prompt_templates.values())
+        return list(self.prompt_templates.values())
 
-    @classmethod
-    def get_prompt_template_names(cls) -> List[str]:
+    def get_prompt_template_names(self) -> List[str]:
         """
         Returns the list of supported prompt template names.
         :return: List of supported prompt template names.
         """
-        return list(cls.prompt_templates.keys())
+        return list(self.prompt_templates.keys())
 
-    @classmethod
-    def is_supported_template(cls, prompt_template: Union[str, PromptTemplate]) -> bool:
+    def is_supported_template(self, prompt_template: Union[str, PromptTemplate]) -> bool:
         """
         Checks if a prompt template is supported.
         :param prompt_template: the prompt template to be checked.
         :return: True if the prompt template is supported, False otherwise.
         """
         template_name = prompt_template if isinstance(prompt_template, str) else prompt_template.name
-        return template_name in cls.prompt_templates
+        return template_name in self.prompt_templates
 
-    @classmethod
-    def get_prompt_template(cls, prompt_template_name: str) -> PromptTemplate:
+    def get_prompt_template(self, prompt_template_name: str) -> PromptTemplate:
         """
         Returns a prompt template by name.
         :param prompt_template_name: the name of the prompt template to be returned.
         :return: the prompt template object.
         """
-        if prompt_template_name not in cls.prompt_templates:
+        if prompt_template_name not in self.prompt_templates:
             raise ValueError(f"Prompt template {prompt_template_name} not supported")
-        return cls.prompt_templates[prompt_template_name]
+        return self.prompt_templates[prompt_template_name]
 
-    @classmethod
-    def prompt_template_params(cls, prompt_template: str) -> List[str]:
+    def prompt_template_params(self, prompt_template: str) -> List[str]:
         """
         Returns the list of parameters for a prompt template.
         :param prompt_template: the name of the prompt template.
         :return: the list of parameters for the prompt template.
         """
-        if not cls.is_supported_template(prompt_template):
+        if not self.is_supported_template(prompt_template):
             raise ValueError(
-                f"{prompt_template} not supported, please select one of: {cls.get_prompt_template_names()}"
+                f"{prompt_template} not supported, please select one of: {self.get_prompt_template_names()}"
             )
 
-        return list(cls.prompt_templates[prompt_template].prompt_params)
+        return list(self.prompt_templates[prompt_template].prompt_params)
+
+    @classmethod
```

Comment:
This looks weird to me - we're not using `cls` so why a class method at all?

In terms of semantic, I would ask myself:
- is `PromptNode` responsible for defining the list? 
  - If no, go with a free function at the module level
  - if yes, does the list need to be accessed elsewhere? 
    - If no, go with a private method
    - if yes, do you need to access `cls`?
        - if yes, go with a `@classmethod`
        - if no, go with a `@staticmethod`



I would go with either a `@staticmethod`, a private method or even better just a free function in the same module. At the end of the day, defining what is a list of predefined templates looks like something outside `PromptNode`'s responsibilities.

Reply:
Ok, good points. Will follow your algorithm and report back. Thanks @masci

Reply:
@masci while at it, wdyt we provide prompt templates as an optional init parameter? If not specified, we'll use the free function at the module level?